src/lib/hooks/useSavedCourses.ts

```diff
@@ -155,6 +157,95 @@ export const useSavedCourses = (): SavedCourses => {
     [contains, containsColor, data, setData]
   );
 
+  /**
+   * Adds a list of timetable courses to the saved courses.
+   * @param courses: List of timetable courses
+   * @param term: Timetable term
+   */
+  const importCourses = useCallback(
+    (courses: TimetableCourse[], term: Term) => {
+      // filtered courses keeps only those not currently saved
+      const filteredCourses: SavedCourse[] = [];
+
+      courses.forEach(({ subject, code, pid, lecture, lab, tutorial, color }) => {
+        // avoid adding a course if it is saved already.
+        const containsCourse =
+          contains(pid, term) || filteredCourses.some((course) => course.pid === pid && course.term === term);
+        if (!containsCourse) {
+          // TODO: Clean up?
+          // I know this looks bad but I tried some different things and none of them were working but this did :S
+          const new_color =
+            !containsColor(color, term) &&
+            !filteredCourses.some((course) => course.color === color && course.term === term)
+              ? color
+              : '#A0AEC0';
+
+          const newCourse: SavedCourse = {
+            subject,
+            pid,
+            code,
+            term,
+            selected: true,
+            showSections: true,
+            lecture: lecture ? lecture[0] : undefined,
+            lab: lab ? lab[0] : undefined,
+            tutorial: tutorial ? tutorial[0] : undefined,
+            color: new_color,
+          };
+          filteredCourses.push(newCourse);
+        }
+      });
+
+      setData([...data].concat(filteredCourses));
+
+      // TODO: add reject
```

Comment:
I copied most of this code from `addCourse` function and decided to leave the comments in case it would be useful to add in the future lol maybe this doesn't apply here